core/src/main/java/org/bitcoinj/coinjoin/CoinJoinClientSession.java

```diff
@@ -110,6 +112,7 @@ public class CoinJoinClientSession extends CoinJoinBaseSession {
 
     private final AtomicBoolean hasNothingToDo = new AtomicBoolean(false); // is mixing finished?
 
+    private final HashMap<Sha256Hash, Integer> collateralSessionMap = new HashMap<>();
```

Comment:
This map will link collateral tx's to the session id, which will make it possible to link a collateral fee tx back to a session.  It could be that the session is `CoinJoinClientSession` has been reset by the time the TX is received.